src/views/LimitOrders/hooks/useGelatoLimitOrdersHistory.ts

```diff
@@ -13,14 +13,21 @@ function newOrdersFirst(a: Order, b: Order) {
   return Number(b.updatedAt) - Number(a.updatedAt)
 }
 
-function syncOrderToLocalStorage({ chainId, account, orders }) {
+function syncOrderToLocalStorage({ chainId, account, orders }: { chainId: number; account: string; orders: Order[] }) {
   const ordersLS = getLSOrders(chainId, account)
 
-  orders.forEach((order: Order) => {
-    const orderExists = ordersLS.find((confOrder) => confOrder.id.toLowerCase() === order.id.toLowerCase())
+  const newOrders = orders.filter(
+    (order: Order) => !ordersLS.some((confOrder) => confOrder.id.toLowerCase() === order.id.toLowerCase()),
+  )
+  saveOrders(chainId, account, newOrders)
+
+  ordersLS.forEach((confOrder: Order) => {
```

Comment:
We can't do batch update here since there updateAt check as well as removing the order when it is status changed

Reply:
Hmm. Got it. Thanks